Assets/__Scripts/MapEditor/Mapping/PlacementControllers/PlacementController.cs

```diff
@@ -150,26 +154,63 @@ public void OnInitiateClickandDrag(InputAction.CallbackContext context)
             if (Physics.Raycast(dragRay, out RaycastHit dragHit, 999f, 1 << 9))
             {
                 BeatmapObjectContainer con = dragHit.transform.gameObject.GetComponent<BeatmapObjectContainer>();
-                if (con is null || !(con is BOC)) return; //Filter out null objects and objects that aren't what we're targetting.
-                isDraggingObject = true;
-                draggedObjectData = con.objectData as BO;
-                originalQueued = BeatmapObject.GenerateCopy(queuedData);
-                originalDraggedObjectData = BeatmapObject.GenerateCopy(con.objectData) as BO;
-                queuedData = BeatmapObject.GenerateCopy(draggedObjectData);
-                draggedObjectContainer = con as BOC;
+                if (StartDrag(con))
+                {
+                    isDraggingObject = true;
+                }
             }
         }
         else if (context.canceled && isDraggingObject && instantiatedContainer != null)
         {
-            //First, find and delete anything that's overlapping our dragged object.
-            objectContainerCollection.RemoveConflictingObjects(new[] { draggedObjectData });
+            FinishDrag();
             isDraggingObject = false;
-            queuedData = BeatmapObject.GenerateCopy(originalQueued);
-            BeatmapActionContainer.AddAction(new BeatmapObjectModifiedAction(draggedObjectData, originalDraggedObjectData, "Modified via alt-click and drag."));
-            ClickAndDragFinished();
         }
     }
 
+    public void OnInitiateClickandDragatTime(InputAction.CallbackContext context)
+    {
+        if (context.performed && CanClickAndDrag)
+        {
+            Ray dragRay = mainCamera.ScreenPointToRay(mousePosition);
+            if (Physics.Raycast(dragRay, out RaycastHit dragHit, 999f, 1 << 9))
+            {
+                BeatmapObjectContainer con = dragHit.transform.gameObject.GetComponent<BeatmapObjectContainer>();
+                if (StartDrag(con))
+                {
+                    isDraggingObjectAtTime = true;
+                    noteGridTransform.position = new Vector3(noteGridTransform.position.x, noteGridTransform.position.y, con.gameObject.transform.position.z);
```

Comment:
Change the `noteGridTransform` reference to the interface scaling offset for each grid (`Rotating/Note Grid/Note Interface Scaling Offset`, and then `Rotating/Event Grid/Event Interface Scaling Offset` for the event grid, etc.), and change its local position instead of global position.